### PR TITLE
feat: add JSON array/object expansion functions

### PR DESCRIPTION
**Pull request by AmanGIT07**
## Why

DWH request: analysts need to expand JSON arrays into rows to analyze workflow specs in `dexter_public.job` (iterate `components[]`, extract AOI ids, map block linkages). Scalar JSON functions (`json_extract`, `->`, `->>`) already work because DuckDB has them natively, but Postgres' set-returning JSON functions have **no DuckDB equivalent — DuckDB has no `json_each`/`json_tree` in any version**, so upgrading the engine was never an option. This implements them as DuckDB macros, following the existing pg-compat macro pattern.

## What's added

| Function | SELECT-list | FROM clause (incl. lateral) |
|---|---|---|
| `json_array_elements` / `jsonb_array_elements` | ✅ | ✅ |
| `json_array_elements_text` / `jsonb_array_elements_text` | ✅ | ✅ |
| `json_object_keys` / `jsonb_object_keys` | ✅ | ✅ |
| `json_each` / `jsonb_each` / `json_each_text` / `jsonb_each_text` | — (STRUCT results panic the wire serializer) | ✅ |

Each name is created as both a scalar macro (`unnest`-based, for `SELECT json_array_elements(...)`) and a table macro (for `FROM json_array_elements(...)`) — DuckDB stores these in separate catalog namespaces, same trick as the existing `pg_is_in_recovery()`.

Query rewriter changes:
- Expressions inside FROM-clause and `JOIN [LATERAL]` function arguments are now remapped (previously skipped entirely), so `::jsonb` casts and schema-qualified column refs work there.
- JOINs are now remapped at **any** position in the FROM list (`FROM a, b JOIN c ON ...`). Previously only `FromClause[0]` was checked, so tables/functions inside later JOINs silently hit the empty DuckDB mirror tables instead of `iceberg_scan` — wrong results, not even an error.

Example (verified against a real workflow spec):

```sql
SELECT j.id, c.value ->> 'name' AS component, d.value AS depends_on
FROM dexter_public.job j,
     jsonb_array_elements(j.spec -> 'components') c,
     jsonb_array_elements_text(c.value -> 'depends_on') d;
```

## Behavior change

`json_object_keys`/`jsonb_object_keys` previously returned the whole key array in a single row; they now return **one row per key**, matching Postgres `SETOF` semantics.

## Pre-existing bugs fixed (found because they blocked or crashed the test suite / server)

1. **`jsonb_extract_path_text(json, VARIADIC ARRAY[...])`** was never remapped (the remapper bailed on non-const args), failing in DuckDB — and its own test panicked the harness. The `VARIADIC ARRAY[...]::text[]` cast form is also supported now.
2. **Crash-by-query class (server process death):** six unguarded `.GetSval().Sval` accesses nil-panicked the per-connection goroutine on non-string constants — `jsonb_extract_path_text(j, 1)`, `1::regclass`, `123::text[]`, `col::text[]`, `123::regproc`, `format(123, ...)`, `bemidb_last_synced_at(123)`, `1 = ANY(123)`. All now degrade to clean SQL errors or graceful fallbacks, verified over the wire.
3. **Test harness panics**: count/type mismatches indexed out of range, aborting the whole suite at a random point (Go randomizes test order). This had been hiding every failure listed below.

## Test status — full transparency

⚠️ **GitHub Actions has never run on this fork** (0 workflow runs; Actions is disabled by default on forks). `build.yml` exists but no PR was ever actually gated by tests. Recommend enabling Actions.

With the harness fixed, the full suite runs to completion. **6 pre-existing failures remain, byte-for-byte identical with and without this branch's changes** (verified by diffing failure lists of both runs):

- `PG_system_tables`: `pg_extension` oid expected `1262`, got `1531`
- `Column_types`: 4× `xid`/`xid8` wire type expected `28`/`5069`, got `20`
- `CASE`: `WHERE nsp.oid = 1268::OID` returns 0 rows

These are hardcoded DuckDB-internal catalog oids/types that drifted with earlier pixxel changes — all reproduce on clean `pixxel` HEAD (they also all exist on upstream `main`, where they pass — so a pixxel change shifted them). Out of scope here; needs a team decision on updating expectations.

The 22 new tests all pass: 19 in the `JSON functions` group (macros in both positions, lateral joins against a synced Iceberg table, `CROSS/LEFT JOIN LATERAL`, JOIN-after-comma positions, `::jsonb` casts, NULL/empty-array handling) plus 3 crash-class regressions in `PG functions`.

## Known limitations

- `WITH ORDINALITY` is not supported by DuckDB — use `row_number() OVER ()` to number elements.
- **DuckDB 1.1.3 engine bug** (reproduced in pure DuckDB, no BemiDB involved): `FROM t, json_fn(...) c LEFT JOIN LATERAL <anything referencing c> ...` fails with `INTERNAL Error: Failed to bind column reference` and **invalidates the DuckDB instance until restart**. Workaround: chain laterals with commas (`FROM t, fn(...) c, fn(c.value...) d` — works fine) or `LEFT JOIN LATERAL` directly off the base table only. Worth a runbook note for the affected instances.

## Verification

- Capabilities probed against the exact bundled engine (go-duckdb v1.8.3 / DuckDB 1.1.3) before implementation.
- Full test suite comparison run: failure set identical with/without the changes.
- End-to-end over the wire: built the binary from this branch, ran a live instance, and executed the real workflow-spec analytics via `psql` (component expansion, nested `inputs[]` AOI extraction, `depends_on` edges, data-flow edges, block-usage rollups, a 15-condition edge-case matrix, the crash-class hostile queries, and the bug-fix repros).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/query_remapper.go [hunk 1]
```diff
@@ -381,6 +390,17 @@ func (remapper *QueryRemapper) remappedExpressions(node *pgQuery.Node, indentLev
 	return remapper.remapperExpression.RemappedExpression(node)
 }
 
+// FROM FUNCTION(args) -> remap expressions in args (e.g. '...'::jsonb casts, schema.table.column references)
+func (remapper *QueryRemapper) remapTableFunctionArgs(rangeFunction *pgQuery.RangeFunction, indentLevel int) {
```
Comment (rohilsurana) [inline src/query_remapper.go]: Minor, fine as a follow-up: these args end up remapped twice. `RemapTableFunctionCall` already runs `RemapFunctionCall` + `RemapNestedFunctionCalls` on the same function calls, and this helper then walks the args through `remappedExpressions`, which remaps nested function calls again.

Harmless today because every function remap renames the function, so the second pass is a no-op. But it is redundant work and a trap if a future remap is not idempotent. Since the two calls are paired at all three call sites, consider folding this into `RemapTableFunctionCall`.

File: src/query_remapper_function.go [hunk 2]
```diff
@@ -110,6 +117,20 @@ func CreatePgCatalogMacroQueries(config *Config) []string {
...
 			'can be bare label' AS baredesc
 		FROM duckdb_keywords()`,
+
+		// JSON set-returning functions used in the FROM clause (incl. laterally): FROM table, json_each(table.column)
+		// Same names as the scalar macros above: DuckDB stores scalar and table macros in separate catalog namespaces
+		"CREATE MACRO json_array_elements(js) AS TABLE SELECT unnest(CAST(js::JSON AS JSON[])) AS value",
+		"CREATE MACRO jsonb_array_elements(js) AS TABLE SELECT unnest(CAST(js::JSON AS JSON[])) AS value",
+		"CREATE MACRO json_array_elements_text(js) AS TABLE SELECT unnest(CAST(js::JSON AS VARCHAR[])) AS value",
+		"CREATE MACRO jsonb_array_elements_text(js) AS TABLE SELECT unnest(CAST(js::JSON AS VARCHAR[])) AS value",
+		"CREATE MACRO json_object_keys(js) AS TABLE SELECT unnest(json_keys(js)) AS json_object_keys",
```
Comment (rohilsurana) [inline src/query_remapper_function.go:127]: Test coverage gap: the FROM-clause (table macro) variants of `json_object_keys`/`jsonb_object_keys`, `jsonb_each`, and `json_each_text` have no direct test. The suite covers `json_each` and `jsonb_each_text` in FROM, and `json_object_keys` only as a scalar. One test each would be cheap and would catch a typo in these macro strings, since they are only parsed at startup.

**Review by rohilsurana — COMMENTED** [review]
Reviewed and verified the main claims on this branch. No blockers.

**What I checked**
- Ran the full test suite on this branch. It runs to completion and fails with exactly the 6 pre-existing failures listed in the description (1 `pg_extension` oid, 4 `xid`/`xid8` wire types, 1 CASE oid lookup). All 13 new JSON tests pass.
- Probed the new macros against the bundled engine (go-duckdb / DuckDB 1.1.3) for edge cases. All match Postgres:
  - `json_each('{"a":1,"a":2}')` returns both rows. The MAP cast handles duplicate keys.
  - Non-array input to `json_array_elements` raises a conversion error, same as Postgres.
  - JSON `null` becomes SQL NULL in the `_text` variants.
  - Object key order is preserved.
- Checked every caller of the nil-safe helper changes (`FirstArgumentToString`, `ArgStringValue`, `MakeListValueFromArray`). All call sites handle the empty/nil results.
- Confirmed the fork has 0 Actions runs. +1 on enabling Actions so `build.yml` gates future PRs.

**Please update the PR description**
The "known limitations" section says the `FROM a, b JOIN c ON ...` gap is unchanged. The last commit fixed it: the JOIN remap now runs over every FROM item, and the new test `FROM (VALUES (1)) v (x), public.test_table t1 JOIN public.test_table t2 ...` proves it works. Please drop or reword that bullet.

**Notes for deploy (not code problems)**
- `json_object_keys` now returns one row per key instead of one array row. Correct per Postgres, but anything built on the old behavior will break. Worth a heads-up to users of the affected instances.
- The DuckDB 1.1.3 `LEFT JOIN LATERAL` bug you documented invalidates the instance until restart, and analysts can now hit it from a psql prompt. Worth a runbook note with the comma-chaining workaround.

File: src/query_remapper.go (hunk 1 above)
Comment (AmanGIT07) [reply src/query_remapper.go]: Done in ca79805 — the pair is folded into a single `remapTableFunction` helper called from all three sites, so the pairing can't drift. The deeper double-dispatch (nested FuncCall args remapped by both `RemapNestedFunctionCalls` and `remappedExpressions`) is endemic to `remappedExpressions` itself (its FuncCall branch re-dispatches everywhere: WHERE, SELECT list, …), so a true single-pass needs a remapper-wide refactor — leaving that as the follow-up.

File: src/query_remapper_function.go (hunk 2 above)
Comment (AmanGIT07) [reply src/query_remapper_function.go:127]: Added in ca79805: FROM-clause tests for `jsonb_each`, `json_each_text`, and both `json[b]_object_keys` table macros (each asserting a specific key/value through the full wire path). All four pass; every macro string in this block now has at least one direct FROM-clause test.

**Review by rohilsurana — APPROVED** [review]
